src/Containers/Microsoft.NET.Build.Containers/Registry/Registry.cs

```diff
@@ -129,31 +133,48 @@ public async Task<ImageBuilder> GetImageManifestAsync(string repositoryName, str
         };
     }
 
-    private async Task<ImageBuilder> ReadSingleImageAsync(string repositoryName, ManifestV2 manifest, CancellationToken cancellationToken)
+    private async Task<ImageBuilder> ReadSingleImageAsync(string repositoryName, string reference, ManifestV2 manifest, RuntimeGraph ridGraph, string applicationRid, CancellationToken cancellationToken)
     {
         cancellationToken.ThrowIfCancellationRequested();
         ManifestConfig config = manifest.Config;
         string configSha = config.digest;
 
         JsonNode configDoc = await _registryAPI.Blob.GetJsonAsync(repositoryName, configSha, cancellationToken).ConfigureAwait(false);
+        var imageConfig = new ImageConfig(configDoc);
+
+        // decipher the RID from the platform information, and then use the deciphered RID to verify compatibility
+        // with the application's targeted RID.
+
+        var configRid = CreateRidForPlatform(imageConfig.PlatformInformation);
+        if (configRid is null)
+        {
+            throw new UnsupportedPlatformException(repositoryName, reference, imageConfig.PlatformInformation);
+        }
+        var configRids = new[] { configRid };
+        var bestManifestRid = GetBestMatchingRid(ridGraph, applicationRid, configRids);
+        if (bestManifestRid is null)
+        {
+            throw new BaseImageNotFoundException(applicationRid, repositoryName, reference, configRids);
+        }
```

Comment:
And here we are, checking compatibility to help give the user some guardrails.

Reply:
I think this guardrail makes sense for almost everyone, but should there be a way to disable it? I'm worried about breaking someone who "knows what they're doing".

Reply:
This is a great point - I should find a way to translate this error into a warning-with-code instead of a hard-stop.